src/server/adminBro.ts

```diff
@@ -162,23 +169,12 @@ export const getAdminBroRouter = async () => {
   return AdminBroExpress.buildAuthenticatedRouter(
     await getAdminBroInstance(),
     {
-      authenticate: async (email, password) => {
-        try {
-          const user = await findAdminUserByEmail(email);
-          if (user) {
-            const matched = await bcrypt.compare(
-              password,
-              user.encryptedPassword,
-            );
-            if (matched) {
-              return user;
-            }
-          }
-          return false;
-        } catch (e) {
-          logger.error({ e });
+      authenticate: async (email, password): Promise<User | boolean> => {
+        const isPasswordCorrect = await checkAdminPassword({ email, password });
+        if (!isPasswordCorrect) {
           return false;
         }
+        return (await findAdminUserByEmail(email)) as User;
```

Comment:
While I agree with the refactor, we are doing the same query twice. We should only query the user once.
Maybe the checkAdminPassword can return the user and we return it here.